central/reports/scheduler/schedule.go

```diff
@@ -411,6 +451,39 @@ func (s *scheduler) runPaginatedQuery(ctx context.Context, scopeQuery, cveQuery
 	return resultData, nil
 }
 
+func (s *scheduler) getDeploymentIDs(ctx context.Context, deploymentsQuery *v1.Query, offset int32) ([]string, error) {
+	deploymentsQuery.Pagination = &v1.QueryPagination{
```

Comment:
Fetch deployments by one deterministic order and keep the responsibility here, with the caller. Do not rely on default ordering in the searcher.